documentation/Architecture/The-Data-Processing-Pipeline.md

```diff
@@ -1,43 +1,131 @@
 # The Data Processing Pipeline
 
-The Custom Data Source (CDS) uses a Data Processor to read from the Data Source. Since the plugins are dynamically loaded into the Driver by the SDK,
-we want to ensure that the Data Processors are efficient in reading and formatting the raw data.
+Within a plugin, a `CustomDataSource` delegates the task of processing data sources to a `CustomDataProcessor`. 
+Because data sources are typically files on disk that need to be both read and parsed, we want to ensure that these 
+processors are efficient and that access to the physical disk occurs as infrequently as possible. Furthermore, it would be 
+nice if the SDK provided a way to
+1) Allow plugins to systematically transform raw data points into an arbitrary data type
+2) Containerize these transformations into specific modules that exist outisde of the `CustomDataProcessor`
+3) Use the output of these transformations in mulitple locations within these modules' plugin (e.g. *share* the output)
+4) Allow external binaries, such as other plugins, access to the output of these transformations (as in *extend* the plugin defining the transformation)
 
+The SDK allows a plugin to achieve these goals, while minimizing the number of times data sources need to be read, 
+by facilitating and allowing the creation of a __data processing pipeline__.
 
-Using the Data Processor, we can efficiently access information from multiple binaries and reuse data.
-Data can be easily shared and the developer is granted maximal flexibility for modular code.
+# Pipeline Components
 
-A Data Processor contains Source Parsers and a Data Cookers.
+At the highest level, when a plugin wishes to create a data processing pipeline, it will define two types of 
+components: __source parsers__ and __data cookers__.
 
-![](.attachments/DataProcessingPipeline.png)
+## Source Parsers
 
+A `SourceParser` is a class whose sole purpose is to process (parse) the raw data inside the data sources a 
```

Comment:
This sentence is super confusing. Maybe try to work off of this so its  a little simpler.
"A SourceParser is a class with the main objective of parsing the raw data from the data source.  Conceptually, a CustomDataProcessor uses the standardized format from the SourceParser and can even delegate the task of processing its data sources to a SourceParser."